backgroundScript.js

```diff
@@ -31,3 +31,43 @@ chrome.runtime.onInstalled.addListener(() => {
 			.then((data) => dataFromAPI = data); // Store the response data in the dataFromAPI variable.
 	})
 })
+
+/**
+ * Checks Logged in status for showing indicator on supported pages like github and bitbucket.
+ * @param {string} websiteUrl
+ * @param {string} provider
+ * @returns {Promise<boolean>}
+ */
+async function checkLoginStatus(websiteUrl, provider) {
+	let result = false;
+	try {
+		const res = await fetch(`${websiteUrl}/api/auth/session`, { cache: 'no-store' });
```

Comment:
> Can you please clarify why you had to go through all the trouble of creating this function in backgroundScript instead of directly making the `fetch` call in the content script itself?
> 
> Were you getting a CORS error?

I was not getting the correct response from fetch when the request is made from the website. I was getting a 200 response but with an empty object.

Reply:
> I am pointing this out because historically, we have found the `message` feature in browser extensions to be unreliable.

I believe some kind of update needs to be done in the backend side if this has to work from content script. Still this could be better security wise I guess, the session data coming to the extension.